tenprintcover.py

```diff
@@ -498,42 +504,68 @@ def drawShape(c, x, y, s):
         else:
             assert not "Implement."
 
-
+    # If the text is long, use a smaller font size. 
+    def scale_font(text, font_name, font_properties):
+        (font_size, font_slant, font_weight) = font_properties
+        width = len(text) * font_size
+        if width > cover_width * 3:   #This is an empirical, unintelligent, heuristic.
+            return  (font_size * 0.8, font_slant, font_weight)
+        elif width < cover_width :
+            return  (font_size * 1.2, font_slant, font_weight)
+        else:
+            return font_properties
+    
+    # Return a font appropriate for the text. Uses Noto CJK if text contains letters of
+    # Simplified Chinese, Traditional Chinese, Japanese, and Korean (CJK), otherwise Noto Sans. 
+    # http://www.unicode.org/faq/han_cjk.html
+    def select_font(text):
+        for char in text:
+            if ord(char) >= 0x4E00:
+                return 'Noto Sans CJK SC'
+        return 'Noto Sans'
+    
     # Allocate fonts for the title and the author, and draw the text.
     def drawText():
         fill = Image.colorRGB(50, 50, 50)
 
         title_font_size = cover_width * 0.08
+        subtitle_font_size = cover_width * 0.05
         title_font_properties = (title_font_size, cairo.FONT_SLANT_NORMAL, cairo.FONT_WEIGHT_BOLD)
-        title_font = cover_image.font("Avenir Next", title_font_properties)
+        subtitle_font_properties = (subtitle_font_size, cairo.FONT_SLANT_NORMAL,
+                                     cairo.FONT_WEIGHT_NORMAL)
+        title_font_family = select_font(title)
+        subtitle_font_family = select_font(subtitle)
+        title_font_properties = scale_font(title, title_font_family, title_font_properties)
+        subtitle_font_properties = scale_font(subtitle, subtitle_font_family, subtitle_font_properties)
+        title_font = cover_image.font(title_font_family, title_font_properties)
+        subtitle_font = cover_image.font(subtitle_font_family, subtitle_font_properties)
         title_height = (cover_height - cover_width - (cover_height * cover_margin / 100)) * 0.75
 
         x = cover_height * cover_margin / 100
         y = cover_height * cover_margin / 100 * 2
         width = cover_width - (2 * cover_height * cover_margin / 100)
         height = title_height
-        cover_image.text(title, x, y, width, height, fill, title_font)
+        title_lines = cover_image.text(title, x, y, width, height, fill, title_font)
+        if subtitle:
+            y = min( y + cover_image.font_height * title_lines * cover_height, 
```

Comment:
I think that [`pylint`](https://www.pylint.org/) will nag about spacing here, have you checked? (Always good to run `pylint` over code, I use it as a check-in requirement.)